.github/workflows/codespell.yml

```diff
@@ -0,0 +1,19 @@
+name: Codespell
+
+on:
+  pull_request:
+
```

Comment:
You need to specify the branches for which the action should trigger.
Like:
```
on:
  pull_request:
    branches:
      - '*'
 ```

Or, 

```
on:
  pull_request:
    branches:
      - 'main'
 ```